src/main/java/org/elasticsearch/action/admin/indices/mapping/get/TransportGetFieldMappingsIndexAction.java

```diff
@@ -133,6 +135,11 @@ protected GetFieldMappingsResponse newResponse() {
         return new GetFieldMappingsResponse();
     }
 
+    @Override
+    protected ClusterBlockException checkRequestBlock(ClusterState state, InternalRequest request) {
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_READ, request.concreteIndex());
+    }
+
```

Comment:
I created #10521 for this, to give this issue some more visibility. Fix is still fine in this PR, just add Closes #10521 to your commit message.